caffe2/ideep/operators/conv_fusion_op.cc

```diff
@@ -75,25 +73,10 @@ class IDEEPConvFusionOp final : public IDEEPConvPoolOpBase {
         "*",
         group_);
 
-    bool weights_changed =
```

Comment:
Why do we remove this?

Reply:
The filter has been converted to optimized format already in optimizeForIdeep.
Nothing needs to do here for inference.

Reply:
This means we have to run optimizeForIdeep in our flow. I prefer that we keep this, because if you have run optimization, this is a no-op but if we somehow missed it, it will implicitly incur large overhead.

Reply:
@yinghai Since the weight format optimization can be applied to both training and inference, another option is that we optimize the format in the default path other than explicitly in optimizeForIdeep() function. Then we do not need to check the "weights_changed" explicitly in convolution op without incurring the "large overhead". What do you think?

Reply:
I meant the function preConvertFiltersFormat() used for weight format optimization.

Reply:
Please don't remove this. I understand with the transformation, this step can be saved but it's not guaranteed that user will run transformation before running the nets. Opt is really an optional step.

Reply:
@yinghai The concern is that this weight cache required the change of the API that creates the convolution operator, i.e. the "training_mode" flag which is off by default. Would this introduce the compatibility issue and require additional changes of the models for training? These models have to explicitly flag "training_mode" on, right?

Reply:
There is no compatibility issue right now as IDEEP training path is brand-new. On the other hand, relying on optimization pass like this before doing inference introduces backward compatibility issue as previously we don't need this.

Reply:
@yinghai I don't insist we should remove this in this PR. We can keep this weight cache for now. But I do think the API for creating models should be the same regardless of the backend devices the models run with, not a problem of whether a backend is new or not. With this "training_mode" flag, users have to explicitly modify their training models so as to run on MKL-DNN optimized backend, right?